src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly

```diff
@@ -34,6 +34,37 @@
             <p:console-link/>
             <l:task icon="images/24x24/notepad.png" href="${buildUrl.baseUrl}/configure" title="${h.hasPermission(it,it.UPDATE)?'%Edit Build Information':'%View Build Information'}"/>
             <st:include page="delete.jelly"/>
+            <j:if test="${it.isBuilding() and h.hasPermission(it,it.CANCEL)}">
+                <j:choose>
+                    <j:when test="${it.hasStopBeenCalled() == false}">
+                        <l:task icon="images/24x24/edit-delete.png"
+                                href="${buildUrl.baseUrl}/stop"
+                                title="${%Stop Build}"
+                                contextMenu="false"
+                                post="true"
+                                requiresConfirmation="true"
+                                confirmationMessage="Really stop build?"/>
+                    </j:when>
+                    <j:when test="${it.hasTermBeenCalled() == false}">
+                        <l:task icon="images/24x24/edit-delete.png"
+                                href="${buildUrl.baseUrl}/term"
+                                title="${%Abort Running Steps}"
+                                contextMenu="false"
+                                post="true"
+                                requiresConfirmation="true"
+                                confirmationMessage="Really forcibly abort running steps?"/>
+                    </j:when>
+                    <j:when test="${it.hasKillBeenCalled() == false}">
```

Comment:
No, we do _not_ want to display the `kill` link instantly after `term` has been called. Similarly, we do not want to display the `term` link instantly after `stop` has been called. The 15s delay is deliberate, to give the build a chance to stop cleanly.